### ISSUE TITLE
`statement_indentation` is too aggressive since 7384 PR

### ISSUE THREAD

**Issue by mvorisek** [PHP-CS-Fixer/PHP-CS-Fixer#7497]
## Feature request

#7384 introduced support for `comment for continuous control statement`. As not everyone up for it, we can introduce rule configuration to opt in/out

----------

Original content: 

## Bug report

introduced in #7384

the PR removed/updated several important tests like:

![image](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/assets/2228672/6db3f3b4-40fb-4627-acf7-13584878a6fd)

In short, it cannot be assumed:

```
if ($v) {
    // todo
}
```

comment only in `if` body is fine but

```diff
-perfectly fine code
+output after 7384 PR
 if ($v) {
     $u = 5;
-    // todo
+// todo
 } else {}
```

with comment at the end it is not. Comments below code can be still contextually valid and relevant to the previous code.

**Comment by Wirone** [comment]
Please look at #7454, I agree with you but unfortunately @keradus not.

**Comment by mvorisek** [comment]
There are at least 4 unique people who want this to be fixed. Currently, the fix can change meaning, so we had to disable this rule...

Please keep this issue open. Either me or someone else can propose the fix.

**Comment by keradus** [comment]
@mvorisek I'm OK to have it open as feature request to have configurable option for the rule.
I'm not OK to change the behaviour fully opposite direction (again, ppl complain regardless which direction one goes with)

@Wirone , sometimes we have not enough code reviews, also not enough contributors. I dare to let myself permission to merge even if there is no review.
